### PR TITLE
PROGRAMMES-6165 - Broadcast Event Object

### PR DESCRIPTION
**Pull request by felipeparaujo**
This also includes quite a bit of the guide page, we just need the series collapsible object here

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/DsShared/Helpers/BroadcastNetworksHelper.php [hunk 1]
```diff
@@ -105,6 +121,17 @@ public function buildServicesNames(array $networkBreakdowns): array
         return $serviceNames;
     }
 
+    private function buildUrlKeys(array $networkBreakdowns): array
```
Comment (Tasiobg) [inline src/DsShared/Helpers/BroadcastNetworksHelper.php]: I'm assuming this will be used in a future, if so please ignore comment

Comment (felipeparaujo) [reply src/DsShared/Helpers/BroadcastNetworksHelper.php]: nope, i just forgot to push the commit that removes it lol

File: src/Controller/ProgrammeEpisodes/GuideController.php [hunk 2]
```diff
@@ -42,6 +49,19 @@ public function __invoke(
             }
         }
 
+        $upcomingBroadcastCount = $collapsedBroadcastService->countUpcomingByProgramme($programme, CacheInterface::MEDIUM);
+        $totalAvailableEpisodes = $programmeAggregationService->countStreamableOnDemandEpisodes($programme);
```
Comment (votemike) [inline src/Controller/ProgrammeEpisodes/GuideController.php]: Are all streamable episodes not on-demand? And vice versa?

Comment (BPScott) [reply src/Controller/ProgrammeEpisodes/GuideController.php]: Things you can watch/listen to online == Available on demand == Streamable.

Editorial wording tends to talk about things being "available on demand", while the field in our DB that says if something is available on iplayer is "isStreamable", which is where the two names for the same concept came from.

IMO `countStreamableOnDemandEpisodes` should just be `countStreamableEpisodes` as it's saying the same thing twice but I didn't want to grump about naming things after a PR was merged :)

Comment (votemike) [reply src/Controller/ProgrammeEpisodes/GuideController.php]: I ❤️ grumping

**Pull request merged**